### ISSUE TITLE
Spatial decomposition - isolate working in confusing way for ifcspaces

### ISSUE THREAD

**Issue by Krande** [IfcOpenShell/IfcOpenShell#5738]
Hey,

I made a simple IFC with 4 box geometries inside an IfcSpace container using scripts outside of Blender. 

After loading the IFC in bonsai, I was kindof expecting the box geometries to be located under the "MySpaces" IfcSpace element in the Scene Collection, not its parent element "MySite" (see the attached example file space_boxes.zip). 

This is the structure I was expecting: 

```
- MySite
    - MySpaces
        - box1
        - box2
        - box3
        - box4
```

![image](https://github.com/user-attachments/assets/46dfaf42-655f-4a2f-b5b4-f6bbd2f81b95)

Is it me misunderstanding the concept of IfcSpace or a bug in the scene collection handling?

For reference, it seems like the "Element Decomposition" menu shows the correct hierarchy relationships.
![image](https://github.com/user-attachments/assets/42a1884e-f956-48b7-98db-029e03e6278b)

I am using bonsai-v0.8.1-alpha241112-26e63c8 and blender 4.2.3 LTS. 

[space_boxes.zip](https://github.com/user-attachments/files/17715928/space_boxes.zip)

Best Regards
Kristoffer

**Comment by Andrej730** [comment]
Hi! Blender collections hierarchy no longer represent the elements decomposition, see https://community.osarch.org/discussion/comment/21943/#Comment_21943. For element decomposition refer to spatial decomposition UI.

**Comment by Krande** [comment]
@Andrej730 I see. Shouldn't then clicking "isolate" in what I assume is the spatial decomposition UI then show the elements reciding within it? (ie. the 4 boxes)?

![isolate_spatial_decomp_ui](https://github.com/user-attachments/assets/66aaaa33-f8ae-4600-9337-c6b0d6e43a24)

**Comment by Andrej730** [comment]
Isolate really only toggles the collections visiblity, it doesnt change objects visibility. So it is correct but confusing. 
But I guess the only way to make it work intuitively (without recreating the spatial hierarchy in the outliner) would be to reset objects visibliy set by user. 

Related to 3c51c63.

**Comment by Krande** [comment]
Okay, so if I understand correctly toggling on/off visibility of containers and child elements does not work for IfcSpace? (which would make sense given that elements contained in the ifcspace are in the scene collection located inside the parent element of IfcSpace).

That's at least the behaviour I am seeing in the clip below. Where you can see selecting IfcSite/IfcBuildingStorey successfully isolates the visibility of the contained elements, but IfcSpace does not.


![compare_storey_vs_space_spatial_ui](https://github.com/user-attachments/assets/1ffa95bb-b8f5-450a-bd62-31653a2c7c88)

I have slightly modified the previous example

```
- MySite
    - MySpace
        - SpaceBox1
        - SpaceBox2
        - SpaceBox3
        - SpaceBox4
     - MyStorey
        - StoreyBox1
        - StoreyBox2
        - StoreyBox3
        - StoreyBox4
```

[space_boxes_space_vs_storey.zip](https://github.com/user-attachments/files/17718527/space_boxes_space_vs_storey.zip)

**Comment by Andrej730** [comment]
> Okay, so if I understand correctly toggling on/off visibility of containers and child elements does not work for IfcSpace? (which would make sense given that elements contained in the ifcspace are in the scene collection located inside the parent element of IfcSpace).

Yes, it's the same problem - collections hierarchy doesn't represent spatial hierarchy, ifcspaces don't have their own collections, it's toggling only collections

**Comment by sboddy** [comment]
The isolate/show/hide underwent quite a change and now works in a way that means that highlighting "MySpaces" and clicking the isolate button (now moved to the pane below) results in isolating those objects contained in the space as @Krande expects.

So I'm closing this bug as it is resolved.
